comp/core/workloadmeta/collectors/internal/kubelet/kubelet.go

```diff
@@ -446,10 +445,20 @@ func extractResources(spec *kubelet.ContainerSpec) workloadmeta.ContainerResourc
 
 	// extract GPU resource info from the possible GPU sources
 	uniqueGPUVendor := make(map[string]bool)
-	for _, gpuResource := range kubelet.GetGPUResourceNames() {
-		if gpuReq, found := spec.Resources.Requests[gpuResource]; found {
-			resources.GPURequest = pointer.Ptr(uint64(gpuReq.Value()))
-			uniqueGPUVendor[extractSimpleGPUName(gpuResource)] = true
+
+	resourceKeys := make([]kubelet.ResourceName, 0, len(spec.Resources.Requests))
+	for resourceName := range spec.Resources.Requests {
+		resourceKeys = append(resourceKeys, resourceName)
+	}
+
+	for _, gpuResourcePrefix := range kubelet.GetGPUResourcePrefixes() {
+		for _, resourceKey := range resourceKeys {
+			if strings.HasPrefix(string(resourceKey), string(gpuResourcePrefix)) {
+				if gpuReq, found := spec.Resources.Requests[resourceKey]; found {
+					resources.GPURequest = pointer.Ptr(uint64(gpuReq.Value()))
+					uniqueGPUVendor[extractGPUVendor(gpuResourcePrefix)] = true
```

Comment:
quick optimization, you could break one of the for loop when you add an item in the `uniqueGPUVendor` map.